src/components/ComputeModal.js

```diff
@@ -1527,6 +1527,9 @@ export const ComputeModalBase = ({
         updateComputeConfig('componentGatewayEnabled', isDataproc(newRuntimeType))
         const machineType = getDefaultMachineType(isDataproc(newRuntimeType), desiredTool)
         updateComputeConfig('masterMachineType', machineType)
+        if (isDataproc(newRuntimeType) && computeConfig.masterDiskSize < defaultDataprocMasterDiskSize) {
+          updateComputeConfig('masterDiskSize', defaultDataprocMasterDiskSize)
```

Comment:
In my opinion, this type of logic should really be in a backend migration. Eventually this 'if' statement will never be true.

Reply:
when will it not be true? what would the migration do?

Reply:
I can see @n8landolt 's point - once everything is migrated (i.e. all disk sizes are increased ) this is essentially dead code. But we will likely increase the default disk size again some time and this code would handle that case. I'm also not a fan of bulk database updates.